src/database_lmdb.h

```diff
@@ -0,0 +1,74 @@
+/*
+Minetest
+Copyright (C) 2013 celeron55, Perttu Ahola <celeron55@gmail.com>
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#ifndef DATABASE_LMDB_HEADER
+#define DATABASE_LMDB_HEADER
+
+#include <vector>
+#include <string>
+#include "irr_v3d.h"
+#include "irrlichttypes.h"
+#include "database.h"
+#include "settings.h"
+#include "lmdb.h"
+
+class Database_LMDB : public Database
+{
+public:
+	Database_LMDB(const std::string &savedir, Settings &conf);
+	~Database_LMDB();
+
+	void beginSave();
+	void endSave();
+
+	bool saveBlock(const v3s16 &pos, const std::string &data);
+	void loadBlock(const v3s16 &pos, std::string *block);
+	bool deleteBlock(const v3s16 &pos);
+
+	void listAllLoadableBlocks(std::vector<v3s16> &dst);
+
+	bool initialized();
+private:
+	MDB_env *environment;
+	MDB_txn *transaction;
+	MDB_dbi database;
+
+	const static size_t defaultMapSize = 2147483648;
```

Comment:
2GB causes a compiler warning on 32 bit. Please use 2GB-1. Also, on 32 bit, 2GB is much to large anyway, as the database is limited by virtual memory, and minetest needs some too. For 32-bit systems, IMO the limit should probably be about 0.5GB below the virtual memory limit of 2GB to 3GB (depending on the system).

From [this](http://www.openldap.org/lists/openldap-technical/201303/msg00145.html) and [this](http://stackoverflow.com/questions/33508305/lmdb-maximum-size-of-the-database-for-windows) (see comment), it seems that just like for Linux, the windows file grows according to need (even if it isn't sparse). Therefore the size can be as large as possible.